apps/web/src/components/planning/Occupation.tsx

```diff
@@ -6,14 +6,27 @@ import { Paginator } from '../generic/Paginator';
 import { OccupationSelector } from '../OccupationSelector';
 import { Form, Formik } from 'formik';
 import { usePlanningContext } from '@services';
-import { useEffect } from 'react';
+import { createContext, useContext, useEffect, useState } from 'react';
 import { useFormikContext } from 'formik';
 
 export interface OccupationProps {
   step: number;
   title: string;
 }
 
+interface OccupationSearchContent {
+  occupationSearch: string;
+  setOccupationSearch: (s: string) => void;
+}
+
+export const OccupationSearchContext = createContext<OccupationSearchContent>({
+  occupationSearch: '',
+  setOccupationSearch: () => {
+    ('');
+  },
+});
+export const useOccupationSearchContext = () => useContext(OccupationSearchContext);
```

Comment:
I don't see a need for context here. It's an overkill. Only the occupationSearch is passed and it's passed to one component which is it's direct child.